starter_code/src/movies.js

```diff
@@ -1,19 +1,177 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+
+function ratesAverage(arr) {
+  const sum = arr.reduce(function(accumulator, currentValue) {
+    let number = parseFloat(currentValue.rate, 10);
+
+    return accumulator + number;
+  }, 0);
+
+  return Math.round((sum / arr.length) * 100) / 100;
+  //always need to return sth in a function while a reduce is used. if not, printing the function is just undefined.
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(arr) {
+  const arrCopy = JSON.parse(JSON.stringify(arr));
+  const dramaMovies = arrCopy.filter(function(element) {
+    if (element.genre.indexOf("Drama") !== -1) {
+      return true;
+    }
+    //return element.genre.includes("Drama");
+  });
+
+  if (dramaMovies.length == 0) {
+    return 0;
+  }
 
+  const sumOfrate = dramaMovies.reduce(function(
```

Comment:
After filtering the array, you could have just used the ratesAverage function here. It would have saved you quite some time :)